fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/WarmupSelectCommand.java

```diff
@@ -0,0 +1,213 @@
+import org.apache.doris.thrift.TQueryStatistics;
+import org.apache.doris.thrift.TQueryStatisticsResult;
+import org.apache.doris.thrift.TUniqueId;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * File cache warm up select command, such as "warm up select * from table where ..."
+ */
+public class WarmupSelectCommand extends InsertIntoTableCommand {
+
+    private static final long QUERY_STATISTICS_TIMEOUT_MS = 5000; // 5 seconds
+    private static final long QUERY_STATISTICS_INTERVAL_MS = 1000; // 1 second
+
+    /**
+     * WarmupSelectCommand constructor
+     */
+    public WarmupSelectCommand(LogicalPlan logicalQuery) {
+        super(PlanType.INSERT_INTO_BLACKHOLE_COMMAND, logicalQuery, Optional.empty(), Optional.empty(),
+                Optional.empty(), true, Optional.empty());
+        super.setInsertExecutorListener(new InsertExecutorListener() {
+
+            @Override
+            public void beforeComplete(AbstractInsertExecutor insertExecutor, StmtExecutor executor, long jobId)
+                    throws Exception {
+                // no-ops
+            }
+
+            @Override
+            public void afterComplete(AbstractInsertExecutor insertExecutor, StmtExecutor executor, long jobId)
+                    throws Exception {
+                if (insertExecutor.ctx.getState().getStateType() == QueryState.MysqlStateType.ERR) {
+                    return;
+                }
+                Map<Long, TQueryStatisticsResult> statistics = pollForQueryStatistics(insertExecutor.ctx.queryId(),
+                        QUERY_STATISTICS_TIMEOUT_MS, QUERY_STATISTICS_INTERVAL_MS);
+                sendAggregatedBlackholeResults(statistics, executor);
+            }
+        });
+    }
+
+    /**
+     * Poll for query statistics until the query is finished or timeout is reached.
+     * Wait for the latest stats from each BE until all BEs have reported isFinished or overall timeout.
+     *
+     * @param queryId The query ID to poll for
+     * @param timeoutMs Maximum time to wait in milliseconds
+     * @param intervalMs Time between polls in milliseconds
+     * @return Map of backend ID to query statistics result
+     */
+    private Map<Long, TQueryStatisticsResult> pollForQueryStatistics(TUniqueId queryId,
+                                                                     long timeoutMs,
+                                                                     long intervalMs) {
+        long startTime = System.currentTimeMillis();
+        Map<Long, TQueryStatisticsResult> beIdToStatisticsResult;
+
+        // Get the coordinator to know how many backends are involved
+        CoordInterface coor = QeProcessorImpl.INSTANCE.getCoordinator(queryId);
+        int expectedBackendCount = 0;
+        if (coor != null) {
```

Comment:
better return directly if `coor == null` or `expectedBackendCount == 0`, to make code more readable